database/src/desktop/util_desktop_macos.mm

```diff
@@ -30,8 +66,17 @@
     int retval;
     retval = mkdir(path.c_str(), 0700);
     if (retval != 0 && errno != EEXIST) return "";
-    retval = mkdir((path + "/" + app_name).c_str(), 0700);
-    if (retval != 0 && errno != EEXIST) return "";
+    // App name might contain path separators. Split it to get list of subdirs
+    std::vector<std::string> app_name_parts = split_string(app_name, '/');
+    if(app_name_parts.empty()) return "";
+    std::string dir_path = path;
+    // Recursively create entire tree of directories
+    for (std::vector<std::string>::const_iterator it = app_name_parts.begin();
+                                            it != app_name_parts.end(); it++) {
+          dir_path = dir_path + "/" + *it;
```

Comment:
This tabbing looks off, as if there's one too many indents.

Reply:
Done!